crates/node/primitives/src/client/blob.rs

```diff
@@ -70,7 +73,233 @@ impl NodeClient {
+                    tracing::warn!(
+                        "Failed to delete metadata for blob {}: {}",
+                        current_blob_id,
+                        err
+                    );
+                }
+            }
+        }
+
+        if deleted_metadata_count > 0 {
+            tracing::info!(
+                "Successfully deleted {} blob metadata entries and {} blob files",
+                deleted_metadata_count,
+                deleted_files_count
+            );
+            Ok(true)
+        } else {
+            bail!("Failed to delete any blob metadata");
+        }
+    }
+
+    /// Get blob metadata
+    ///
+    /// Returns blob metadata including size, hash, and detected MIME type.
+    /// This is efficient for checking blob existence and getting metadata info.
+    pub async fn get_blob_info(&self, blob_id: BlobId) -> eyre::Result<Option<BlobMetadata>> {
+        let handle = self.datastore.clone().handle();
+        let blob_key = BlobMeta::new(blob_id);
+
+        match handle.get(&blob_key) {
+            Ok(Some(blob_meta)) => {
+                let mime_type = self
+                    .detect_blob_mime_type(blob_id)
+                    .await
+                    .unwrap_or_else(|| "application/octet-stream".to_owned());
+
+                Ok(Some(BlobMetadata {
+                    blob_id,
+                    size: blob_meta.size,
+                    hash: blob_meta.hash,
+                    mime_type,
+                }))
+            }
+            Ok(None) => Ok(None),
+            Err(err) => {
+                tracing::error!("Failed to get blob metadata: {:?}", err);
+                bail!("Failed to retrieve blob metadata: {}", err);
+            }
+        }
+    }
+
+    /// Detect MIME type by reading the first few bytes of a blob
+    pub async fn detect_blob_mime_type(&self, blob_id: BlobId) -> Option<String> {
+        match self.get_blob(&blob_id) {
+            Ok(Some(mut blob_stream)) => {
+                if let Some(Ok(first_chunk)) = blob_stream.next().await {
+                    let bytes = first_chunk.as_ref();
+                    let sample_size = std::cmp::min(bytes.len(), 512); // Read more bytes for better detection
+                    return Some(detect_mime_from_bytes(&bytes[..sample_size]));
+                }
+            }
+            Ok(None) => {
+                tracing::warn!("Blob {} not found for MIME detection", blob_id);
+            }
+            Err(err) => {
+                tracing::warn!(
+                    "Failed to read blob {} for MIME detection: {:?}",
+                    blob_id,
+                    err
+                );
+            }
+        }
+
+        None
+    }
+}
+
+/// Detect MIME type from file bytes using the infer crate
+fn detect_mime_from_bytes(bytes: &[u8]) -> String {
+    if let Some(kind) = infer::get(bytes) {
+        return kind.mime_type().to_owned();
+    }
+
+    "application/octet-stream".to_owned()
```

Comment:
`mime_type` already returns a `&'static str`

```suggestion
/// Detect MIME type from file bytes using the infer crate
fn detect_mime_from_bytes(bytes: &[u8]) -> &'static str {
    if let Some(kind) = infer::get(bytes) {
        return kind.mime_type();
    }

    "application/octet-stream"
```